rsciio/delmic/_api.py

```diff
@@ -1,80 +1,804 @@
+                    f"is outside of expected ranges."
+                )
+
+            # Handle the special cases for the C, T, and A axes
+            if axis_name == "C":
+                axis_dict = {
+                    "name": "Wavelength",
+                    "axis": Scale * unit_factor(prefix),  # Full axis loaded
+                    "units": prefix + "m",  # Adjusted unit prefix
+                    "navigate": False,  # No navigation for the C axis
+                }
+            elif axis_name == "T":
+                axis_dict = {
+                    "name": "Time",
+                    "axis": Scale * unit_factor(prefix),  # Full axis loaded
+                    "units": prefix + "s",  # Adjusted unit prefix
+                    "navigate": False,  # No navigation for the T axis
+                }
+            elif axis_name == "A":
+                ScaleA = np.arange(0, Image.shape[1])
+                axis_dict = {
+                    "name": "Angle",
+                    "axis": ScaleA,  # Full axis loaded, no scaling needed
+                    "units": "",  # Unit for Angle
+                    "navigate": False,  # No navigation for the A axis
+                }
+            else:
+                axis_dict = {
+                    "name": axis_name,
+                    "size": Image.shape[
+                        -(i + 1)
+                    ],
+                    "offset": float(np.array(ImgData.get(offset_key)))
+                    * unit_factor(prefix),
+                    "scale": scale_value * unit_factor(prefix),
+                    "units": prefix + "m",
+                    "navigate": True,
+                }
+            axes.append(axis_dict)
+
+    # Reorder axes to X, Y, Wavelength, Angle
+    axes_ordered = []
+    for axis_name in ["Y", "X", "Time", "Angle", "Wavelength"]:
+        for axis in axes:
+            if axis["name"] == axis_name:
+                axes_ordered.append(axis)
+                break
+
+    return axes_ordered
+
+
+def make_metadata(Acq):
+    """Create a metadata dictionary for a given acquisition (Acq)."""
+    metadata = {
+        "Signal": {
+            "quantity": "",  # Placeholder, to be filled with actual data
+            "signal_type": "",  # Placeholder, to be filled with actual data
+            "signal_origin": "",  # Placeholder, to be filled with actual data
+        }
+    }
+    if "quantity" in Acq:
+        metadata["Signal"]["quantity"] = Acq["quantity"]
+    else:
+        metadata["Signal"]["quantity"] = "Intensity (counts)"  # Default value
+
+    if "signal_type" in Acq:
+        metadata["Signal"]["signal_type"] = Acq["signal_type"]
+    else:
+        metadata["Signal"]["signal_type"] = ""  # Default value
+
+    if "signal_origin" in Acq:
+        metadata["Signal"]["signal_origin"] = Acq["signal_origin"]
+    else:
+        metadata["Signal"]["signal_origin"] = ""  # Default value
+
+    return metadata
+
+
+def make_original_metadata(Acq):

```

Comment:
`original_metadata` should be 1:1 the metadata tree from the file and can be automatically parsed for hdf5 files, see e.g.

https://github.com/hyperspy/rosettasciio/blob/332412f93b27c014311a84b13eea9852dfb2707f/rsciio/nexus/_api.py#L828

`metadata` should contain a curated subset of metadata with predefined key names that follows the LumiSpy/HyperSpy conventions:
https://docs.lumispy.org/en/stable/user_guide/metadata_structure.html

Reply:
As a note, HyperSpy's `load` function takes the parameter `load_original_metadata` that can be used to only parse the curated `metadata` and drop the object `original_metadata` in case it is very large:
https://hyperspy.org/hyperspy-doc/current/user_guide/io.html#metadata